pkg/deploy/registry/deploylog/rest_test.go

```diff
@@ -0,0 +1,175 @@
+		list.Items = append(list.Items, makeDeployment(v))
+	}
+	return list
+}
+
+// Mock pod resource getter
+type deployerPodGetter struct{}
+
+func (p *deployerPodGetter) Get(ctx kapi.Context, name string) (runtime.Object, error) {
+	return &kapi.Pod{
+		ObjectMeta: kapi.ObjectMeta{
+			Name:      name,
+			Namespace: kapi.NamespaceDefault,
+		},
+		Spec: kapi.PodSpec{
+			Containers: []kapi.Container{
+				{
+					Name: name + "-container",
+				},
+			},
+			NodeName: name + "-host",
+		},
+	}, nil
+}
+
+// mockREST mocks a DeploymentLog REST
+func mockREST(version, desired int, endStatus api.DeploymentStatus) *REST {
+	// Fake deploymentConfig
+	fakeDn := testclient.NewSimpleFake(deploytest.OkDeploymentConfig(version))
+	// Fake deployments
+	fakeDeployments := makeDeploymentList(version)
+	fakeRn := ktestclient.NewSimpleFake(fakeDeployments)
+	// Fake watcher for deployments
+	fakeRn.Watch = watch.NewFake()
+	// Everything is fake
+	connectionInfo := &kclient.HTTPKubeletClient{Config: &kclient.KubeletConfig{EnableHttps: true, Port: 12345}, Client: &http.Client{}}
+
+	go func(obj *kapi.ReplicationController) {
+		// Add deployment in the fake watcher - status New
+		fakeRn.Watch.(*watch.FakeWatcher).Add(obj)
+		time.Sleep(100 * time.Millisecond)
+
+		// Modify deployment in the fake watcher - status Pending
+		obj.Annotations[api.DeploymentStatusAnnotation] = string(api.DeploymentStatusPending)
+		fakeRn.Watch.(*watch.FakeWatcher).Modify(obj)
+		time.Sleep(100 * time.Millisecond)
+
+		// Modify deployment in the fake watcher - status depends on endStatus
+		// Running, Complete, or Failed, all should be tested
+		obj.Annotations[api.DeploymentStatusAnnotation] = string(endStatus)
+		fakeRn.Watch.(*watch.FakeWatcher).Modify(obj)
+	}(&fakeDeployments.Items[desired-1])
+
+	return &REST{
+		DeploymentConfigsNamespacer:      fakeDn,
+		ReplicationControllersNamespacer: fakeRn,
+		PodGetter:                        &deployerPodGetter{},
+		ConnectionInfo:                   connectionInfo,
+		Timeout:                          defaultTimeout,
+	}
+}
+
+func TestRESTGet(t *testing.T) {
+	ctx := kapi.NewDefaultContext()
+
+	tests := []struct {
+		testName    string
+		rest        *REST
+		name        string
+		opts        runtime.Object
+		expected    runtime.Object
+		expectedErr error
+	}{
+		{
+			testName: "running deployment",
+			rest:     mockREST(1, 1, api.DeploymentStatusRunning),
+			name:     "config",
+			opts:     &api.DeploymentLogOptions{Follow: true, Version: 1},
+			expected: &genericrest.LocationStreamer{
+				Location: &url.URL{
+					Scheme:   "https",
+					Host:     "config-1-deploy-host:12345",
+					Path:     "/containerLogs/default/config-1-deploy/config-1-deploy-container",
+					RawQuery: "follow=true",
+				},
+				Transport:   nil,
+				ContentType: "text/plain",
+				Flush:       true,
+			},
+			expectedErr: nil,
+		},
+		// TODO: Uncomment once the testclient supports GETing objects that are added by NewSimpleFake
```

Comment:
This will be refactored anyway after the rebase since the testclient has changed quite a bit so I'll leave it as is for now. Commented tests pass if you use this simple fix: https://github.com/kubernetes/kubernetes/pull/12282